src/Interpreters/InterpreterAlterQuery.cpp

```
@@ -59,6 +62,26 @@ namespace ErrorCodes
     extern const int QUERY_IS_PROHIBITED;
 }
 
+namespace
+{
+    void updateDatabaseCommentWithMetadataFile(const DatabasePtr & db, const AlterCommand & command)
+    {
+        if (!command.comment)
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Can not get database comment from query");
+
+        String old_database_comment = db->getDatabaseComment();
+        db->setDatabaseComment(command.comment.value());
+        try
+        {
+            DatabaseCatalog::instance().updateDatabaseComment(db);
+        }
+        catch (...)
+        {
+            db->setDatabaseComment(old_database_comment);
```

We might need to re-throw the exception here